python_modules/dagster/dagster/_core/storage/db_io_manager.py

```diff
@@ -131,6 +133,15 @@ def load_input(self, context: InputContext) -> object:
             context, self._get_table_slice(context, cast(OutputContext, context.upstream_output))
         )
 
+    def _get_partition_value(
+        self, partition_def: PartitionsDefinition, partition_key: str
+    ) -> Union[TimeWindow, str]:
+        try:
```

Comment:
Can we use an `isinstance(partitions_def, TimewindowPartitionsDefinition)` call instead of a try catch block?